src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -496,26 +505,153 @@ void DxFontRenderData::SetFeatures(std::unordered_map<std::wstring_view, uint32_
     _featureVector.clear();
     for (const auto [tag, param] : _featureMap)
     {
-        const auto dwriteTag = DWRITE_MAKE_FONT_FEATURE_TAG(gsl::at(tag, 0), gsl::at(tag, 1), gsl::at(tag, 2), gsl::at(tag, 3));
+        const auto dwriteTag = DWRITE_MAKE_FONT_FEATURE_TAG(til::at(tag, 0), til::at(tag, 1), til::at(tag, 2), til::at(tag, 3));
         _featureVector.push_back(DWRITE_FONT_FEATURE{ dwriteTag, param });
     }
 }
 
 // Routine Description:
 // - Updates our internal map of font axes with the given axes
+// - NOTE TO CALLER: Make sure to call UpdateFont after calling this for the feature changes
+//   to take place
 // Arguments:
 // - axes - the axes to update our map with
 void DxFontRenderData::SetAxes(std::unordered_map<std::wstring_view, int32_t> axes)
 {
-    _axesMap.clear();
+    _axesVector.clear();
 
     // Update our axis map with the provided axes
     for (const auto [axis, value] : axes)
     {
-        _axesMap[axis] = value;
+        if (axis.length() == TAG_LENGTH)
+        {
+            const auto dwriteTag = DWRITE_MAKE_FONT_AXIS_TAG(gsl::at(axis, 0), gsl::at(axis, 1), gsl::at(axis, 2), gsl::at(axis, 3));
+            _axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ dwriteTag, gsl::narrow<float>(value) });
+        }
+    }
+}
+
+// Method Description:
+// - Converts a DWRITE_FONT_STRETCH enum into the corresponding float value to
+//   create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontStretch: the old DWRITE_FONT_STRETCH enum to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontStretch
+float DxFontRenderData::FontStretchToWidthAxisValue(const DWRITE_FONT_STRETCH fontStretch) noexcept
+{
+    if (gsl::narrow<size_t>(fontStretch) > fontStretchEnumToVal.size())

```

Comment:
`gsl::narrow` can technically throw. And you marked this `noexcept`. If it's a value that you're 100% certain will not throw or can be truncated to no ill-effect, use `gsl::narrow_cast`